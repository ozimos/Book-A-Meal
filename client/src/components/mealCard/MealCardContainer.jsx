import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';

import '../../../public/styles/book_a_meal.css';


const MealCardContainer = ({ MealCard, meals, ...props }) =>
  (
    <div className={props.addClass ? `${props.addClass} gallery` : 'gallery'}>
      {meals.map(meal => (<MealCard
        key={meal.id}
        {...meal}
        {...props}
      />))
        }
    </div>
  );
MealCardContainer.defaultProps = {
  addClass: PropTypes.string
};
MealCardContainer.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  MealCard: PropTypes.func.isRequired,
  addClass: PropTypes.string,
};

export default hot(module)(MealCardContainer);

