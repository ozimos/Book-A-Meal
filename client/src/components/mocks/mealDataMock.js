export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
