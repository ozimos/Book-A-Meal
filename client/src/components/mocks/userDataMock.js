export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
