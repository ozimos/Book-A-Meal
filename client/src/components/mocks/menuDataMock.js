import { allMeals } from './mealDataMock';

export default {
  data: {
    rows: [{
      id: '42858b92-dfb4-4859-98be-7d89765bb3fe',
      menuDate: '2018-08-28',
      createdAt: '2018-08-28T14:11:50.210Z',
      updatedAt: '2018-08-28T14:11:50.210Z',
      Meals: allMeals.data.rows
    }] }
};
