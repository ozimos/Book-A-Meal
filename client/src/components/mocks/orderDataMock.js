import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};