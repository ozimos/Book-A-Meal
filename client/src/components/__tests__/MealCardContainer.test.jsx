/* global React:false, shallow:false toJson:false */

import MealCardContainer from '../mealCard/MealCardContainer';
import { rows } from '../mocks/mealDataMock';

const props = {
  meals: rows,
  MealCard: jest.fn(),
};

describe('MealCardContainer Component', () => {
  const setup = () => shallow(<MealCardContainer {...props} />);
  it('renders correctly', () => {
    let wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
    const props2 = { ...props, addClass: 'string' };
    const setup2 = () => shallow(<MealCardContainer {...props2} />);
    wrapper = setup2();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
