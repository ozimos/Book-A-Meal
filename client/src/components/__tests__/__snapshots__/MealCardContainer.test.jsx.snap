// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCardContainer Component renders correctly 1`] = `
<div
  className="gallery"
>
  <mockConstructor
    addClass=""
    id="abc"
    key="abc"
    price={1200}
  />
  <mockConstructor
    addClass=""
    id="def"
    key="def"
    price={2000}
  />
</div>
`;

exports[`MealCardContainer Component renders correctly 2`] = `
<div
  className="string gallery"
>
  <mockConstructor
    addClass="string"
    id="abc"
    key="abc"
    price={1200}
  />
  <mockConstructor
    addClass="string"
    id="def"
    key="def"
    price={2000}
  />
</div>
`;
